Refine inbox styling and formatting

olaseni opened the pull request:
#### Changes proposed in this Pull Request

* Update some styles based on comments from pcDL0W-Jl-p2#comment-992

#### Testing instructions 

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Find a site with some mailboxes
* Navigate to /inbox
* Compare style fixes from the comment above

#### Screenshots

##### Desktop

<img width="1087" alt="Screenshot 2021-10-01 at 13 43 55" src="https://user-images.githubusercontent.com/3376401/135615699-41c60aa7-170c-41bb-83f9-faa98553cc85.png">

##### Tablet

<img width="767" alt="Screenshot 2021-10-01 at 13 44 17" src="https://user-images.githubusercontent.com/3376401/135615694-9b639169-942a-429c-ade8-ef8a976accaa.png">

##### Mobile

<img width="375" alt="Screenshot 2021-10-01 at 13 44 50" src="https://user-images.githubusercontent.com/3376401/135615691-c71d4386-54a8-4ef0-8444-73c50c5a044c.png">

olaseni committed:
Add styles for the main container

olaseni committed:
Refine some styles

olaseni commented:
Pulling you in here @saygunnyc because you'll have more context on this.

saygunnyc commented:
> Pulling you in here @saygunnyc because you'll have more context on this.

Thank you, @olaseni! Some key screenshots would mean the world, if possible.

olaseni commented:
Here are some screenshots.
<img width="900" alt="Screenshot 2021-09-30 at 7 14 53 PM" src="https://user-images.githubusercontent.com/277661/135509314-2b120a55-021e-4d82-abd4-1f911e3c864c.png">

<img width="350" alt="Screenshot 2021-09-30 at 7 15 47 PM" src="https://user-images.githubusercontent.com/277661/135509340-389bbcf2-cd4f-4aa9-9165-0d3f4b768c6b.png">

saygunnyc commented:
It looks like we still don't have the correct card sizes on Desktop and Mobile. Should be 85 px in height.
Also, the margins on mobile don't seem right. It should be 16 px. Hope the screenshot shared below is helpful.

<img width="929" alt="Screen Shot 2021-10-01 at 11 40 48" src="https://user-images.githubusercontent.com/78876323/135591389-136bc930-2d98-49bc-a63d-68a7eca364f9.png">

Also, that would be great to reduce the spacing and make it half the size, in between "Need another mailbox" and "Create new and activate immediately" over the Desktop.

<img width="1307" alt="Screen Shot 2021-10-01 at 11 43 57" src="https://user-images.githubusercontent.com/78876323/135591824-8d950f90-d847-4fba-a05d-ffef922d92b7.png">

daledupreez committed:
Ensure icon area doesn't get squished

daledupreez committed:
Width-related fixes
  - Switch to non-flex list items on mobile
  - Remove hard-coded widths for narrower screens above mobile size

daledupreez commented:
@saygunnyc, I've responded to your feedback items and think they've been addressed. I didn't modify the vertical spacing in the create new mailbox section as I think it was sufficiently small already.

LMK if the screenshots look good.

saygunnyc commented:
> LMK if the screenshots look good.
Looking better, thank you!

I don't think we need to change the header and body-copy size on tablet size as we don't seem to change mailbox card sizes.

AllTerrainDeveloper reviewed (approved):
It looks good to me!

### Evidence of Testing
![mailbox-resize](https://user-images.githubusercontent.com/5689927/135636988-9df04756-3372-4081-8d7d-1fb4408d836e.gif)
